src/parser/retail/modules/items/thewarwithin/trinkets/MadQueensMandate.ts

```diff
@@ -0,0 +1,24 @@
+import ITEMS from 'common/ITEMS/thewarwithin/trinkets';
+import SPELLS from 'common/SPELLS/thewarwithin/trinkets';
+import Analyzer, { Options } from 'parser/core/Analyzer';
+import SPELL_CATEGORY from 'parser/core/SPELL_CATEGORY';
+import Abilities from 'parser/core/modules/Abilities';
+
+export default class MadQueensMandate extends Analyzer.withDependencies({
+  abilities: Abilities,
+}) {
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTrinket(ITEMS.MAD_QUEENS_MANDATE.id);
+    if (!this.active) {
+      return;
+    }
+
+    this.deps.abilities.add({
+      spell: SPELLS.ABYSSAL_GLUTTONY.id,
+      category: SPELL_CATEGORY.COOLDOWNS,
+      cooldown: 120,
+    });
+  }
```

Comment:
if you want to add the death logic for MQM, the way i'd do it is:

1. add `SpellUsable` as a dependency.
2. add a `Events.damage` handler to the module
3. if the damage event has overkill, reduce the cooldown by 1 minute. this will sometimes be wrong (some enemies don't actually die), but it will be right way, way more than it is wrong.

the handler would look something like:

```ts
private onDamage(event: DamageEvent) {
  if (event.overkill && event.overkill > 0) {
    this.spellUsable.reduceCooldown(SPELLS.ABYSSAL_GLUTTONY.id, 60);
  }
}
```

if you did something like this in the constructor:

```ts
this.addEventListener(Events.damage.spell(SPELLS.ABYSSAL_GLUTTONY), this.onDamage);
```

Reply:
Thanks for the help. Added the cd logic. Lmk if you need anything else to get this along

Reply:
(i actually got the scale wrong, it should be 60000, not 60 since it is in milliseconds. but i will patch that in the morning and merge)

Reply:
good catch. updated.